### DISCUSSION TITLE
Is there any method to avoid consuming about 5~10 seconds to loading pyodide and other packages?

### DISCUSSION THREAD

**Discussion by dongrixinyu** [pyscript/pyscript#1610] [Q&A]
Is there any method to avoid consuming about 5~10 seconds to loading pyodide and other packages? When I first open the brower.

![image](https://github.com/pyscript/pyscript/assets/27001062/0a965e25-b285-4144-abb9-a5ff149665f2)

**Comment by WebReflection** [comment]
This looks like a question for *Pyodide* team, not *PyScript*, but from our side, be aware you can lazily inject custom script or py-script elements on the page, having that wait time delayed or *on demand* when, and if, needed.

On the other side, specially when you have a config with 3rd party modules and/or files to fetch, the network time plays a role ... it's true that pyodide takes 1 up to 2 seconds to bootstrap itself, and AFAIK they are working on improving some performance there as we speak, but modules and fetched files are either orchestrated or fetched inevitably *before* your code can run, otherwise you couldn't `import` anything from any of those modules.

Maybe one day we'll manage to provide lazy imports so that stuff is loaded only when any of your files needs it, but I don't see that happening any time soon + it would require code in workers to keep it synchronous-but-not-synchronous behind the scene ... which is all doable, but surely not trivial.
